log/provider.go

```diff
@@ -0,0 +1,93 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package log // import "go.opentelemetry.io/otel/log"
+
+import (
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/log/embedded"
+)
+
+// LoggerProvider TODO: comment.
+type LoggerProvider interface {
+	embedded.LoggerProvider
+
+	// Logger TODO: comment.
+	Logger(name string, options ...LoggerOption) Logger
```

Comment:
Are we worried that `options ...LoggerOption` can cause memory allocations?

Personally, I think that it is acceptable as I people should not get a new logger for each HTTP/RPC call.

However, maybe my assumption is wrong?

But even if so then the same problem would be for `TraceProvider` and `MeterProvider`. 

Therefore, I would prefer to keep the options here for the sake of similarity.

Reply:
d4896f64

Reply:
I think some bridge implementation required getting a new Logger for every since record emitted. Can't remember why exactly at the moment. Probably because the logger name can be different for every record.

Reply:
@tigrannajaryan Good catch. I think it could be [`github.com/go-logr/logr.LogSink.WithName`](https://pkg.go.dev/github.com/go-logr/logr#LogSink). I will add some naïve implementations of `logr` and `slog` bridges with benchmarks to evaluate the design. But I assume that it would be better to make `LoggerOptions` as  a regular `struct`.